pootle/core/url_helpers.py

```diff
@@ -12,7 +12,83 @@
 import urllib
 import urlparse
 
-from django.core.urlresolvers import reverse
+from django.core.urlresolvers import reverse, resolve
+from django.utils.functional import cached_property
+
+
+class PathResolver(object):
+    """Converts a request.path into parts for matching pootle_path
+    with a regex
```

Comment:
I would rephrase this:

Normalizes the request.path, extracts its parts and provides a regex useful to match the path.

Or use a even more explanatory docstring with examples.